cloudigrade/util/aws/cloudtrail.py

```diff
@@ -0,0 +1,130 @@
+        session (boto3.Session): A temporary session tied to a customer account
+        aws_account_id (string): The aws account id to include in the name
+        include_global_service_events (bool): Specifies whether trail is
+            publishing events from global services to log files
+        is_multi_region_trail (bool): Specifies whether the trail is created in
+            all regions or current region
+
+    Returns:
+        The response code of creating cloudtrail.
+
+    """
+    cloudtrail = session.client('cloudtrail')
+    name = f'cloudigrade-{aws_account_id}'
+
+    if lookup_trail(cloudtrail, name):
+        response = update_cloudtrail(cloudtrail, name,
+                                     include_global_service_events,
+                                     is_multi_region_trail)
+    else:
+        response = create_cloudtrail(cloudtrail, name,
+                                     include_global_service_events,
+                                     is_multi_region_trail)
+    # limit to write only events & turn on logging
+    put_event_selectors(cloudtrail, name)
+    cloudtrail.start_logging(Name=name)
+    return response
+
+
+def lookup_trail(cloudtrail, name):
+    """
+    Check to see if a cloudigrade CloudTrail exists for the account.
+
+    Args:
+        cloudtrail (botocore client): The cloudtrail client
+        name (string): The name of the cloudtrail to lookup
+    Returns:
+        A bool stating whether or not the trail exists.
+
+    """
+    trails = cloudtrail.describe_trails(trailNameList=[name])
+    # the trailList will not be empty if a trail with the specified name exists
+    return trails.get('trailList', []) != []
+
+
+def put_event_selectors(cloudtrail, name):
+    """
+    Specify that the trail collect data on write-only events.
+
+    Args:
+        cloudtrail (boto3.Session): A temp
+        name (string): The name of the cloudtrail to configure
+    """
+    event_selector = [{'ReadWriteType': 'WriteOnly'}]
+    try:
+        response = cloudtrail.put_event_selectors(
+            TrailName=name,
+            EventSelectors=event_selector
+        )
+        return response
+    except ClientError as e:
+        raise e
+
+
+def create_cloudtrail(cloudtrail, name, include_global_service_events,
+                      is_multi_region_trail):
+    """Create a CloudTrail in the customer account.
+
+    Args:
+        cloudtrail (botocore client): The cloudtrail client
+        name (string): The name of the cloudtrail to configure
+        include_global_service_events (bool): Specifies whether trail is
+            publishing events from global services to log files
+        is_multi_region_trail (bool): Specifies whether the trail is created in
+            all regions or current region
+    """
+    logger.debug(_('Creating the cloudtrail {0}').format(name))
+    try:
+        response = cloudtrail.create_trail(
+            Name=name,
+            S3BucketName=settings.S3_BUCKET_NAME,
```

Comment:
Should this have a namespace for the shared account?  It looks like we all will use `S3_BUCKET_NAME`.  Wondering if it should be `username-S3_BUCKET_NAME`